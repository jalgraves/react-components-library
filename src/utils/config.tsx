// Configs used for testing components
const config = {
  "static_url": "https://static.prod.beantownpub.com",
  "sections": {
    "home": {
      "template": "index",
      "metadata": {
        "title": "Dr. Davis Ice Cream",
        "description": "Doc's Ice Cream Pepperell Massachusetts, making and serving our own ice cream for over 80 years!",
        "keywords": "ice cream, homemade, desserts, pepperell, massachusetts, sundaes, milkshakes, frappes, soft-serve"
      }
    },
    "menu": {
      "template": "menu",
      "metadata": {
        "title": "Doc's Menu",
        "description": "Dr. Davis Ice Cream Menu",
        "keywords": "ice cream, homemade, desserts"
      }
    },
    "about": {
      "template": "about",
      "metadata": {
        "title": "About Doc's",
        "description": "",
        "keywords": ""
      }
    },
    "contact": {
      "template": "contact",
      "metadata": {
        "title": "Contact Doc's",
        "description": "",
        "keywords": ""
      }
    },
    "parties": {
      "template": "parties",
      "metadata": {
        "title": "Ice Cream Parties",
        "description": "",
        "keywords": ""
      }
    },
    "flavors": {
      "template": "flavors",
      "metadata": {
        "title": "Doc's Ice Cream Flavors",
        "description": "",
        "keywords": "chocolate, vanilla, strawberry, coffee"
      }
    },
    "faq": {
      "template": "faq",
      "metadata": {
        "title": "Doc's FAQ",
        "description": "",
        "keywords": ""
      }
    }
  },
  "urls": {
    "home": "https://drdavisicecream.com",
    "googleApi": "https://maps.googleapis.com/maps/api/",
    "static": "https://static.prod.beantownpub.com",
    "facebook": "https://www.facebook.com/Doc-Davis-Ice-Cream-Stand-115542725133788/",
    "instagram": "https://www.instagram.com/drdavisicecream/",
    "twitter": "https://twitter.com/DrDavisIceCream"
  },
  "colors": {
    "yellow": "rgb(252,204,18)",
    "primaryYellow": "rgb(252,204,18)",
    "gray": "#494040",
    "red": "#c70039",
    "primaryRed": "#c70039",
    "navBarBlue": "rgba(5,46,102,0.90)",
    "primaryBlue": "rgba(5,46,102,0.90)",
    "navBarMenuBackgroundGray": "rgb(248 248 248)",
    "backgroundGray": "rgba(248,248,248,0.8)",
    "primaryGray": "rgb(248 248 248)",
    "blue": "#1f6696",
    "lightGray": "#F3F4F4",
    "darkGray": "#888888",
    "cyan": "#3fc5f0",
    "black": "#000000",
    "white": "whitesmoke"
  },
  "fonts": {
  "button": "'gotham-narrow-ultra', Poppins, sans-serif",
  "content": "'gotham-medium', Lato, sans-serif",
  "footer": "'Raleway', Arial, sans-serif",
  "headline": "'Poppins', sans-sefif",
  "script": "'Satisfy', cursive;"
  },
  "pages": {
    "index": {
      "name": "index",
      "icon": "faHome",
      "text": "Home",
      "title": "Dr. Davis Ice Cream",
      "metadata": {
        "title": "Dr. Davis Ice Cream",
        "description": "Doc's Ice Cream Pepperell Massachusetts, making and serving our own ice cream for over 80 years!",
        "keywords": "ice cream, homemade, desserts, pepperell, massachusetts, sundaes, milkshakes, frappes, soft-serve"
      },
      "menuList": true,
      "topMenu": false
    },
    "about": {
      "name": "about",
      "icon": "faInfo",
      "text": "About",
      "title": "About Doc's",
      "description": "Information about Dr. Davis Ice Cream",
      "menuList": true,
      "topMenu": true
    },
    "contact": {
      "name": "contact",
      "icon": "faEnvelope",
      "text": "Contact",
      "title": "Contact Doc's",
      "description": "Contact information for Dr. Davis Ice Cream",
      "metadata": {
        "title": "Ice Cream Parties",
        "description": "",
        "keywords": ""
      },
      "menuList": true,
      "topMenu": true
    },
    "menu": {
      "name": "menu",
      "icon": "faUtensils",
      "text": "Menu",
      "title": "Doc's Menu",
      "description": "Dr. Davis Ice Cream food menu",
      "metadata": {
        "title": "Ice Cream Parties",
        "description": "",
        "keywords": ""
      },
      "menuList": true,
      "topMenu": true
    },
    "parties": {
      "name": "parties",
      "icon": "faUsers",
      "text": "Parties",
      "title": "Ice Cream Parties",
      "description": "Dr. Davis Ice Cream parties",
      "metadata": {
        "title": "Ice Cream Parties",
        "description": "",
        "keywords": ""
      },
      "menuList": true,
      "topMenu": true
    },
    "flavors": {
      "name": "flavors",
      "icon": "faStar",
      "text": "Flavors",
      "title": "Ice Cream Flavors",
      "metadata": {
        "title": "Dr. Davis Ice Cream",
        "description": "Doc's Ice Cream Pepperell Massachusetts, making and serving our own ice cream for over 80 years!",
        "keywords": "ice cream, homemade, desserts, pepperell, massachusetts, sundaes, milkshakes, frappes, soft-serve"
      },
      "description": "Doc's ice cram flavors",
      "menuList": false,
      "topMenu": false
    },
    "facebook": {
      "name": "https://www.facebook.com/DrDavisIceCream/",
      "icon": "faFacebook",
      "text": "Facebook",
      "title": "",
      "metadata": {
        "title": "Dr. Davis Ice Cream",
        "description": "Doc's Ice Cream Pepperell Massachusetts, making and serving our own ice cream for over 80 years!",
        "keywords": "ice cream, homemade, desserts, pepperell, massachusetts, sundaes, milkshakes, frappes, soft-serve"
      },
      "description": "",
      "menuList": true,
      "topMenu": false
    },
    "instagram": {
      "name": "https://www.instagram.com/drdavisicecream/",
      "icon": "faInstagram",
      "text": "Instagram",
      "title": "",
      "metadata": {
        "title": "Dr. Davis Ice Cream",
        "description": "Doc's Ice Cream Pepperell Massachusetts, making and serving our own ice cream for over 80 years!",
        "keywords": "ice cream, homemade, desserts, pepperell, massachusetts, sundaes, milkshakes, frappes, soft-serve"
      },
      "description": "",
      "menuList": false,
      "topMenu": false
    },
    "twitter": {
      "name": "https://twitter.com/DrDavisIceCream",
      "icon": "faTwitter",
      "text": "Twitter",
      "title": "",
      "metadata": {
        "title": "Dr. Davis Ice Cream",
        "description": "Doc's Ice Cream Pepperell Massachusetts, making and serving our own ice cream for over 80 years!",
        "keywords": "ice cream, homemade, desserts, pepperell, massachusetts, sundaes, milkshakes, frappes, soft-serve"
      },
      "description": "",
      "menuList": false,
      "topMenu": false
    }
  }
}

export default config
