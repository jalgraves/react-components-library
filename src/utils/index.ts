import * as configObj from './config'

const config = configObj.default

export { config }
