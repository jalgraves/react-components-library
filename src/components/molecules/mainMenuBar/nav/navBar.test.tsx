import React from "react"
import renderer from "react-test-renderer"
import NavBar from "./navBar"
import { config } from "../../../../utils/index"

it('nav bar snapshot', () => {
  const component = renderer.create(
    <NavBar
      ariaDetails="TestNavBar" 
      fontColor="cyan"
      fontFamily="Arial"
      hoverColor="yellow"
      imgAlt="Dr. Davis Ice Cream Logo"
      imgSource={`${config.static_url}/img/logos/drdavisicecream_navbar_logo.gif`}
      pages={config.pages}
    />
  )
  let tree = component.toJSON() as any
  console.log("NavBar")
  console.log(tree.props)
  console.log(tree.children)
  expect(tree).toMatchSnapshot()
})
