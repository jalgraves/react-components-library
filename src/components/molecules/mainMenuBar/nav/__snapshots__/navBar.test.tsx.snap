// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav bar snapshot 1`] = `
<div
  aria-details="TestNavBar"
  className="sc-beyTiQ bjtWal"
>
  <nav
    aria-details="NavBarLinks"
    className="topNav"
  >
    <a
      href="About"
      id="About-mobile-link"
    >
      About
    </a>
    <a
      href="Contact"
      id="Contact-mobile-link"
    >
      Contact
    </a>
    <a
      href="Menu"
      id="Menu-mobile-link"
    >
      Menu
    </a>
    <a
      href="Parties"
      id="Parties-mobile-link"
    >
      Parties
    </a>
  </nav>
</div>
`;
